tests/e2e/test_rosacli_upgrade.go

```diff
@@ -305,6 +305,120 @@ var _ = Describe("Cluster Upgrade testing",
 		})
 	})
 
+var _ = Describe("Describe/List rosa upgrade",
+	labels.Feature.Cluster, func() {
+		defer GinkgoRecover()
+		var (
+			rosaClient     *rosacli.Client
+			clusterService rosacli.ClusterService
+			upgradeService rosacli.UpgradeService
+			clusterID      string
+			profile        *profilehandler.Profile
+		)
+
+		BeforeEach(func() {
+			By("Get the cluster")
+			clusterID = config.GetClusterID()
+			Expect(clusterID).ToNot(Equal(""), "ClusterID is required. Please export CLUSTER_ID")
+
+			By("Init the client")
+			rosaClient = rosacli.NewClient()
+			clusterService = rosaClient.Cluster
+			upgradeService = rosaClient.Upgrade
+
+			By("Load the profile")
+			profile = profilehandler.LoadProfileYamlFileByENV()
+		})
+
+		AfterEach(func() {
+			By("Delete cluster upgrade")
+			output, err := clusterService.DeleteUpgrade("-c", clusterID, "-y")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(output.String()).To(ContainSubstring("Successfully canceled scheduled upgrade on cluster '%s'", clusterID))
+
+			By("Clean remaining resources")
+			err = rosaClient.CleanResources(clusterID)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("to list/describe rosa upgrade via ROSA CLI - [id:57094]",
+			labels.High, labels.Runtime.Day2,
+			func() {
+				By("Check the help message of 'rosa describe upgrade -h'")
+				output, err := clusterService.DescribeUpgrade(clusterID, "-h")
+				Expect(err).To(BeNil())
+				Expect(output.String()).To(ContainSubstring("rosa describe upgrade [flags]"))
+				Expect(output.String()).To(ContainSubstring("-c, --cluster"))
+				Expect(output.String()).To(ContainSubstring("--machinepool"))
+				Expect(output.String()).To(ContainSubstring("-y, --yes"))
+
+				By("Check list upgrade for the cluster with latest version")
+				if profile.Version == "latest" {
+					output, err = upgradeService.ListUpgrade(clusterID)
+					Expect(err).To(BeNil())
+					Expect(output.String()).To(ContainSubstring("There are no available upgrades for cluster '%s'", clusterID))
+				}
+
+				By("Upgrade cluster and check list/describe upgrade")
+				if profile.Version == con.YStreamVersion {
+					scheduledDate := time.Now().Format("2006-01-02")
+					scheduledTime := time.Now().Add(20 * time.Minute).UTC().Format("15:04")
+
+					jsonData, err := clusterService.GetJSONClusterDescription(clusterID)
+					Expect(err).To(BeNil())
+					clusterVersion := jsonData.DigString("version", "raw_id")
+
+					versionService := rosaClient.Version
+					clusterVersionList, err := versionService.ListAndReflectVersions(profile.ChannelGroup, false)
+					Expect(err).ToNot(HaveOccurred())
+
+					versions, err := clusterVersionList.FindYStreamUpgradeVersions(clusterVersion)
+					Expect(err).To(BeNil())
+					Expect(len(versions)).
```

Comment:
Should we put an error here ?
If there is no available versions, then we have no path to it. Therefore it cannot be tested.
But I would not fail the test

Reply:
if len(versions) == 0, will skip the test instead of fail the test.